vivarium/interface/interactive.py

```diff
@@ -92,11 +92,23 @@ def replace_component(self, old_component, new_component):
         self.component_manager.add_components([new_component])
 
 
-def setup_simulation(components, input_config=None):
+def initialize_simulation(components, input_config=None, plugin_config=None, context_class=InteractiveContext):
     config = build_simulation_configuration()
     config.update(input_config)
 
-    simulation = InteractiveContext(config, components)
+    base_plugin_config = ConfigTree(DEFAULT_PLUGINS['plugins'])
```

Comment:
Since you and Abie are both trying to modify this code to add in some additional flexibility, we should sit down and figure out what the needs around this are.  Both of your solutions feel clunky to me.  Passing around classes as arguments to functions in the top-level interactive API feels user hostile.

Reply:
Agree!